### ISSUE TITLE
[BUG] Fails to start iq scan

### ISSUE THREAD

**Issue by Iletee** [sonatype-nexus-community/auditjs#132]
**Description**
Running auditjs iq results in a crash and burn due to 'could not instantiate muncher'

> Ilkkas-MacBook-Pro:juice-shop7 ilkkaturunen$ auditjs iq -a app123 -s build
 ________  ___  ___  ________  ___  _________      ___  ________
|\   __  \|\  \|\  \|\   ___ \|\  \|\___   ___\   |\  \|\   ____\
\ \  \|\  \ \  \\\  \ \  \_|\ \ \  \|___ \  \_|   \ \  \ \  \___|_
 \ \   __  \ \  \\\  \ \  \ \\ \ \  \   \ \  \  __ \ \  \ \_____  \
  \ \  \ \  \ \  \\\  \ \  \_\\ \ \  \   \ \  \|\  \\_\  \|____|\  \
   \ \__\ \__\ \_______\ \_______\ \__\   \ \__\ \________\____\_\  \
    \|__|\|__|\|_______|\|_______|\|__|    \|__|\|________|\_________\
                                                          \|_________|

✔ Starting application
{ level: 'error',
  message: 'Could not instantiate muncher',
  label: 'AuditJS',
  timestamp: '2020-01-23T09:37:47.087Z' }
/usr/local/lib/node_modules/auditjs/bin/Application/Application.js:70
            throw new Error("Could not instantiate muncher");
            ^

Error: Could not instantiate muncher
    at new Application (/usr/local/lib/node_modules/auditjs/bin/Application/Application.js:70:19)
    at Object.<anonymous> (/usr/local/lib/node_modules/auditjs/bin/index.js:156:19)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3

**Steps to reproduce**

> npm install -g auditjs@beta
> auditjs config
> (configure IQ server settings)
> auditjs iq -a appid -s buld
> also fails on any other auditjs command

**Environment**
OS: MacOs Catalina
npm version: 6.13.6
auditjs: tested from alpha-11 to alpha-14 and 4.0.1-beta.2

**Comment by hboutemy** [comment]
FYI, exact same issue for me (same OS, npm 6.13.4)

**Comment by ajurgenson55** [comment]
Are you guys running it in an npm project?

**Comment by ajurgenson55** [comment]
Means its not passing this check:

```
  public isValid(): boolean {
    let tempPath = path.join(process.cwd(), "package-lock.json");
    return fs.existsSync(tempPath);
  }
```

just want to confirm there's a package-lock.json in the directory you're scanning before troubleshooting the function in your environments

**Comment by ajurgenson55** [comment]
I'll change the console output as well so it's clearer when you're in an invalid directory

**Comment by Iletee** [comment]
Duh - that was it - i thought I was running it in the project root - turns
out i was one directory short. Some more descriptive docs would be good to
prevent this

**Comment by DarthHater** [comment]
@Iletee @ajurgenson55 addressed this in #133 , should get released to beta fairly quickly :)

**Comment by hboutemy** [comment]
I confirm I was not in a js project, just trying to launch first
Now, the message is better:
```✔ Starting application
{
  level: 'error',
  message: 'Failed project directory validation.  Are you in a (built) node, yarn, or bower project directory?',
  label: 'AuditJS',
  timestamp: '2020-01-23T16:15:27.093Z'
}
Could not instantiate muncher```

would even be better without the `Could not instanciate muncher` part, but good enough

**Comment by hboutemy** [comment]
continuing my basic user test: launching `auditjs` on clone of https://github.com/sonatype-nexus-community/search-maven-org
the output is
```
$ auditjs
 ________  ___  ___  ________  ___  _________      ___  ________
|\   __  \|\  \|\  \|\   ___ \|\  \|\___   ___\   |\  \|\   ____\
\ \  \|\  \ \  \\\  \ \  \_|\ \ \  \|___ \  \_|   \ \  \ \  \___|_
 \ \   __  \ \  \\\  \ \  \ \\ \ \  \   \ \  \  __ \ \  \ \_____  \
  \ \  \ \  \ \  \\\  \ \  \_\\ \ \  \   \ \  \|\  \\_\  \|____|\  \
   \ \__\ \__\ \_______\ \_______\ \__\   \ \__\ \________\____\_\  \
    \|__|\|__|\|_______|\|_______|\|__|    \|__|\|________|\_________\
                                                          \|_________|


✔ Starting application
```

surprising end...

**Comment by hboutemy** [comment]
1151 components found in audijs ossi mode
0 component (and hence no violation...) in auditjs iq mode

there is an issue

**Comment by DarthHater** [comment]
@hboutemy there may be, going to need more information on how you ran things, what directory did you run from, etc...

> continuing my basic user test: launching `auditjs` on clone of https://github.com/sonatype-nexus-community/search-maven-org
> the output is
> 
> ```
> $ auditjs
>  ________  ___  ___  ________  ___  _________      ___  ________
> |\   __  \|\  \|\  \|\   ___ \|\  \|\___   ___\   |\  \|\   ____\
> \ \  \|\  \ \  \\\  \ \  \_|\ \ \  \|___ \  \_|   \ \  \ \  \___|_
>  \ \   __  \ \  \\\  \ \  \ \\ \ \  \   \ \  \  __ \ \  \ \_____  \
>   \ \  \ \  \ \  \\\  \ \  \_\\ \ \  \   \ \  \|\  \\_\  \|____|\  \
>    \ \__\ \__\ \_______\ \_______\ \__\   \ \__\ \________\____\_\  \
>     \|__|\|__|\|_______|\|_______|\|__|    \|__|\|________|\_________\
>                                                           \|_________|
> 
> 
> ✔ Starting application
> ```
> 
> surprising end...
> I would have expected the help message, or anything telling there is an issue

addressed in: https://github.com/sonatype-nexus-community/auditjs/commit/e1a9fc51360700d722baccc0deadf6025c3cf2d0 , will make it to beta shortly.

**Comment by ajurgenson55** [comment]
@Iletee fixed the invalid project directory UX with #133 and it's pushed to beta, check it out :)

@hboutemy what's the command you're running for IQ that you're getting 0 components?

something like `auditjs iq -a search-maven-org` will work (if you ran `auditjs config`)

I agree there should be some kind of message is ossi or iq isn't specified, will add it shortly

**Comment by DarthHater** [comment]
@ajurgenson55 the command message missing is fixed in https://github.com/sonatype-nexus-community/auditjs/commit/e1a9fc51360700d722baccc0deadf6025c3cf2d0

@hboutemy going to need more information, after installing `auditjs` like `npm i -g auditjs@beta`, 

```
(base) 503 thedude:search-maven-org (DependencyHell *)$ auditjs iq -a testapp
 ________  ___  ___  ________  ___  _________      ___  ________      
|\   __  \|\  \|\  \|\   ___ \|\  \|\___   ___\   |\  \|\   ____\     
\ \  \|\  \ \  \\\  \ \  \_|\ \ \  \|___ \  \_|   \ \  \ \  \___|_    
 \ \   __  \ \  \\\  \ \  \ \\ \ \  \   \ \  \  __ \ \  \ \_____  \   
  \ \  \ \  \ \  \\\  \ \  \_\\ \ \  \   \ \  \|\  \\_\  \|____|\  \  
   \ \__\ \__\ \_______\ \_______\ \__\   \ \__\ \________\____\_\  \ 
    \|__|\|__|\|_______|\|_______|\|__|    \|__|\|________|\_________\
                                                          \|_________|
                                                                      
                                                                      
✔ Starting application
✔ Getting coordinates for Nexus IQ Server
✔ Auditing your application with Nexus IQ Server
✔ Getting your internal application ID
✔ Submitting your dependencies to Nexus IQ Server
✔ Checking for results (this could take a minute)
Wonderbar! No policy violations reported for this audit!
Report URL: http://localhost:8070/ui/links/application/testapp/report/034e3a3a960f463a9a779393bb0f0ed5
```

The report shows the amount of components I expect to see (23 in this case).

**Comment by Iletee** [comment]
I've opened #134 to address the next issue. My initial problem is solved but leaving open for follow on debugging

**Comment by ajurgenson55** [comment]
Fixed by #133
